hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/common/HoodieSparkEngineContext.java

```diff
@@ -148,6 +148,7 @@ public <I, K, V> Stream<ImmutablePair<K, V>> mapPartitionsToPairAndReduceByKey(
                 .map(e -> new Tuple2<>(e.getKey(), e.getValue())).iterator()
         )
         .reduceByKey(reduceFunc::apply)
+        .coalesce(parallelism)
```

Comment:
Is the intention to change the parallelism of `reduceByKey`?

Reply:
yes, btw, found that reduceKyKey itslef takes an additional arg. I have fixed it.